### PR TITLE
Make actions dropdown closeable.

### PR DESCRIPTION
**Pull request by mathewjordan**
# Summary 
First, a note. This ended up being a larger issue as the way we handled it required an integration of a new component from our design system. So apologies for the partial overhaul here.

This pull request make actions dropdown closeable by replacing it's existing functionality with a Popover component from the design-system repository. As a side effect of bringing in popover, we also needed to make some adjustments to the Technical Metadata and Delete (Work/Fileset) modals to address z-index layering issues. To do this, I have brought in the Dialog component from Radix UI. Long term a modal component should be created for the design system (issue created as [#2489](https://github.com/nulib/repodev_planning_and_docs/issues/2489)), but for now this has been inserted directly from the Radix UI package -- making the length of `Preservation.jsx` a bit longer than it could be.

image

# Specific Changes in this PR
- Swap out actions menu for design-system popover.
- Convert technical metadata modal to Radix UI alert dialog.
# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.

- [x] Patch
- [ ] Minor
- [ ] Major

# Steps to Test

1.  Go to a work with multiple filesets associated.
2. Click Preservation tab
3. Click actions, close the panel by clicking the `X` close button, hitting `esc`, or by clicking outside of the popover pane.
4. Click View technical metadata, a modal should appear, which is also closeable. The modal should disable interaction with outside elements and only popover content will be visible to screen readers until closed.
5. Can delete work and filesets still

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
  - [ ] Database Schema changes
  - [ ] GraphQL API
  - [ ] Elasticsearch API
  - [ ] Ingest Sheet
  - [ ] CSV metadata export/update API
  - [ ] Shared Links export API
- Backwards-incompatible API changes
  - [ ] Database Schema changes
  - [ ] GraphQL API
  - [ ] Elasticsearch API
  - [ ] Ingest Sheet
  - [ ] CSV metadata export/update API
  - [ ] Shared Links export API
- [ ] Requires data migration
- [ ] Requires database triggers disabled during deployment/migration
- [ ] Requires reindex
- [ ] Terraform changes
  - [ ] Adds/requires new or changed Terraform variables
- [ ] Pipeline configuration changes (requires `mix meadow.pipeline.setup` run)
- [ ] Requires new variable added to `miscellany`
- [ ] Specific deployment synchronization instructions with other apps/API's
- [ ] Other specific instructions/tasks

# Tested/Verified
- [ ] End users/stakeholders

### CODE REVIEW

**Review by adamjarling — CHANGES_REQUESTED** [review]
Looks and works great.  A few minor UI tweaks, and also wondering about whether we want to commit to Meadow knowing about `stitches` as a dependency....

image

image

**Comment by adamjarling** [comment]
@mathewjordan Thinking a little more about this, I'd really like Meadow to know nothing about anything other than our Design System in the future.  I'm ok with leaving the direct import of `{ styled }` from Stitches now if we can create an issue to create a "Dialog" or "Modal" component in our NUL Design System repo which will eventually replace the direct Radix Dialog component now introduced into Meadow.

Thoughts?

**Comment by kdid** [comment]
Functionality to close the actions dropdown works great. Noticed same as mentioned above - that I can't scroll the structural metadata content if it's long in the modal. 

About swapping the icons - the icon next to the "Copy preservation location to clipboard" action is an (S3) bucket so I wouldn't make that the icon for delete - there probably is a trash can icon or something more appropriate? I guess if I were going to replace the S3 bucket with the copy to clipboard icon I'd think you'd also replace the "Copy checksums to clipboard" with it. But then all three copies would have the same icon... I kind of think it's fine the way it is, personally.

File: assets/js/components/Work/Tabs/Preservation/Preservation.jsx [hunk 1]
```diff
@@ -26,21 +26,15 @@ import {
...
-  IconDownload,
-  IconTrashCan,
   IconView,
 } from "@js/components/Icon";
-import UIDropdown from "@js/components/UI/Dropdown";
-import UIDropdownItem from "@js/components/UI/DropdownItem";
-import UIIconText from "@js/components/UI/IconText";
-import DownloadAll from "@js/components/UI/Modal/DownloadAll";
+import * as Dialog from "@radix-ui/react-dialog";
+import { styled } from "@stitches/react";
```
[the remark this answers is not in the document]
Comment (mathewjordan) [reply assets/js/components/Work/Tabs/Preservation/Preservation.jsx:32]: I was thinking we would want to change this out after we complete https://github.com/nulib/repodev_planning_and_docs/issues/2489. My goal here was to get things rolling and compatible with Popover first.

File: assets/js/components/Work/Tabs/Preservation/Preservation.jsx [hunk 2]
```diff
@@ -266,58 +252,181 @@ const WorkTabsPreservation = ({ work }) => {
...
+                                      isText
+                                      css={{
+                                        padding: "0",
+                                        justifyContent: "left",
+                                      }}
+                                      onClick={() =>
+                                        handleDeleteFilesetClick(fileset)
+                                      }
+                                    >
+                                      <IconCopyToClipboard />
```
[the remark this answers is not in the document]
Comment (mathewjordan) [reply assets/js/components/Work/Tabs/Preservation/Preservation.jsx:382]: Oh, yes. Whoops.

**Comment by mathewjordan** [comment]
@adamjarling @kdid 
Good catch on the Icon, copy/paste casualty. I just set them back to how there were before.
image

The overflow issue should now be resolved as well.

**Comment by mathewjordan** [comment]
> @mathewjordan Thinking a little more about this, I'd really like Meadow to know nothing about anything other than our Design System in the future. I'm ok with leaving the direct import of `{ styled }` from Stitches now if we can create an issue to create a "Dialog" or "Modal" component in our NUL Design System repo which will eventually replace the direct Radix Dialog component now introduced into Meadow.
> 
> Thoughts?

@adamjarling I've created two issues for this:

Create Modal in Design System
https://github.com/nulib/repodev_planning_and_docs/issues/2489

Replace Dialog/Stitches with Modal in Meadow
https://github.com/nulib/repodev_planning_and_docs/issues/2492

File: assets/js/components/Work/Tabs/Preservation/Preservation.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (adamjarling) [reply assets/js/components/Work/Tabs/Preservation/Preservation.jsx:33]: Totally, get it.

**Review by adamjarling — APPROVED** [review]

**Review by kdid — APPROVED** [review]

**Pull request merged**